components/accelerator/nvidia/infiniband/threshold/threshold.go

```diff
@@ -0,0 +1,122 @@
+// Package threshold contains the threshold for the infiniband component.
+package threshold
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/leptonai/gpud/components"
+	"github.com/leptonai/gpud/components/common"
+	events_store "github.com/leptonai/gpud/components/db"
+)
+
+// Defines the minimum number of ports and the expected rate in Gb/sec.
+type Threshold struct {
+	// The minimum number of ports.
+	// If not set, it defaults to the number of GPUs.
+	AtLeastPorts int `json:"at_least_ports"`
+
+	// The expected rate in Gb/sec.
+	// If not set, it defaults to 200.
+	AtLeastRate int `json:"at_least_rate"`
+}
+
+func (th Threshold) JSON() ([]byte, error) {
+	return json.Marshal(th)
+}
+
+func Parse(b []byte) (*Threshold, error) {
+	var th Threshold
+	err := json.Unmarshal(b, &th)
+	if err != nil {
+		return nil, err
+	}
+	return &th, nil
+}
+
+func (th Threshold) Event(time time.Time) (components.Event, error) {
```

Comment:
I'm curious that why not store these data to a separate file which stores all node hardware spec or node related thresholds?

Reply:
Yes, we can do that as well (separate table might be easier, since we already have all the persistence via sqlite file)

Reply:
OK, but it would be better to use a text file to store this configuration, as it would be more maintainable.